mintpy/save_gbis_mimtpy.py

```diff
@@ -0,0 +1,522 @@
+    parser.add_argument('-m', '--mask', dest='mask_file', help='mask file.')
+    parser.add_argument('--ref-lalo', dest='ref_lalo', type=float, nargs='?',
+                        help='custom reference pixel in lat/lon')
+    
+    
+    parser.add_argument('-outdir','--outdir',dest='outdir',nargs=1,
+                        help='output directory')
+
+    return parser
+
+def cmd_line_parse(iargs=None):
+    parser = create_parser()
+    inps = parser.parse_args(args=iargs)  
+    # read templatefile
+    if inps.templateFile:
+        inps = read_template2inps("".join(inps.templateFile), inps)
+    else:    
+        # default startDate and endDate
+        if not os.path.isfile("".join(inps.file)):
+            file = find_timeseries(os.getcwd())
+        else:
+            file = "".join(inps.file)    
+        atr = readfile.read_attribute(file)
+        if not inps.startDate or inps.startDate=='None':
+            inps.startDate = atr['START_DATE']
+        if not inps.endDate or inps.endDate=='None':
+            inps.endDate = atr['END_DATE']
+    return inps    
+
+
+def read_template2inps(templatefile, inps):
+    """Read input template options into Namespace inps"""
+    print('read input option from template file: ' + templatefile)
+    if not inps:
+        inps = cmd_line_parse()
+    inps_dict = vars(inps)
+
+    template = readfile.read_template(templatefile)    
+    template = ut.check_template_auto_value(template)
+    
+    prefix = 'gbis.'
+    key_list = [i for i in list(inps_dict.keys()) if prefix + i in template.keys()]
+    for key in key_list:
+        value = template[prefix + key]
+        if value:
+            if key == 'DataSet':
+                inps_dict[key] = list(tuple([i for i in value.split(',')]))
+            elif key == 'DataType':
+                inps_dict[key] = value
+            elif key == 'SNWE':
+                inps_dict[key] = list(tuple([float(i) for i in value.split(',')]))
+            elif key in ['latStep', 'lonStep']:
+                inps_dict[key] = float(value)
+            elif key in ['startDate','endDate']:
+                inps_dict[key] = value
+            elif key in ['mask_file']:
+                inps_dict[key] = value
+            elif key in ['ref_lalo']:
+                inps_dict[key] = list(tuple([float(i) for i in value.split(',')]))
+
+    inps.laloStep = [inps.latStep, inps.lonStep]
+    if None in inps.laloStep:
+        inps.laloStep = None
+    print(inps)
+    return inps
+
+def generate_outdir_name(inps,pwdDir):
+    """set output directory"""
+    if not inps.outdir or inps.outdir[0] == 'None':
+        projectTrack = pwdDir.split('/')[-2] #get projectSenAT***
+        ret = re.findall(r"^(.+)(Sen[AD]T\d+)$", projectTrack)
+        project = ret[0][0]
+        Track = ret[0][1]
+        dirname = "".join([os.getenv('MODELDIR')+'/'+project+'/'+Track+'/gbis'+'_'+inps.startDate+'_'+inps.endDate])
```

Comment:
It's not recommended to embed environmental variables in the code as other users might not have it.

We are always try to keep mintpy code independent from RSMAS setting.